### DISCUSSION TITLE
What does TrustedProxies allow to be passed in from the Proxy?

### DISCUSSION THREAD

**Discussion by imjustmatthew** [photoprism/photoprism#3382] [Q&A] [answered]
Hi,

I see that there is a `TrustedProxies` option (`PHOTOPRISM_TRUSTED_PROXY`), but I can't find in the source code what actually consumes this option and what I can do with it. Does it let me pass in authentication data like an authenticated username/e-mail from the proxy?

This is where its updated inline comments are: https://github.com/photoprism/photoprism/commit/c29bc5a8d4c9ef49e7c265fca1338515d0008d64

Thanks!

**Comment by imjustmatthew** [comment]
I see this line also, maybe this is really the only place it gets used?
https://github.com/photoprism/photoprism/blob/60162b3fc510c6cafd7983ad08fee9547b893858/internal/server/start.go#L39

**Reply by lastzero** [reply, marked as answer]
https://pkg.go.dev/github.com/gin-gonic/gin#Engine.SetTrustedProxies

**Reply by imjustmatthew** [reply]
Thanks!

For future searches: The web framework PhotoPrism uses (Gin) only seems to use this to decide whether the trust the X-Forwarded-For headers to determine the real client IP. `PHOTOPRISM_TRUSTED_PROXY` will take multiple CIDR ranges separated by commas. It also supports IPv6 ranges. The parsing is handled by this line: https://github.com/gin-gonic/gin/blob/fe989b6a6f8091b2708b39a60b1dd2a2bd3b2812/gin.go#L438

Interesting lines of Gin about this:

- https://github.com/gin-gonic/gin/blob/fe989b6a6f8091b2708b39a60b1dd2a2bd3b2812/gin.go#L137 where the Remote IP comments are
- https://github.com/gin-gonic/gin/blob/fe989b6a6f8091b2708b39a60b1dd2a2bd3b2812/gin.go#L419 where the IP ranges are processed
